src/java.xml/share/classes/module-info.java

```diff
@@ -27,11 +27,70 @@
  * Defines the Java API for XML Processing (JAXP), the Streaming API for XML (StAX),
  * the Simple API for XML (SAX), and the W3C Document Object Model (DOM) API.
  *
+ * <h2 id="ConfigurationFile">JAXP Configuration File</h2>
+ * JAXP supports the use of a configuration file for the
+ * <a href="#LookupMechanism">Factory Lookup Mechanism</a> and
+ * setting properties that have defined corresponding system properties.
```

Comment:
The first sentence of the module description lists the 4 XML APIs and their acronyms. It is immediately followed by a section with title "JAXP Configuration File" which suggests that this is a configuration file for the first API that is listed (JAXP). As I understand it, properties for the streaming API can also be defined in this file. So it might be that a bit more setup is needed in the opening text to make it clearer that this is the configuration file for all of the APIs.

Reply:
One other thing to say on this is that jaxp.properties has "XML Library (java.xml) Configuration File" at the top of the file, maybe that could be used in the module description instead of "JAXP Configuration File".

Reply:
Changed the opening statement and also the naming of config file as above.

Reply:
I've looked through the updated proposal but I think the intro is going to take a few iterations to get to something that is easy to read. I had hoped the intro would provide an overview of the APIs in the module and establish the terminology that is needed to read further. The existing text isn't great but it does name the APIs in the module so the reader knows there are several APIs for processing XML. With the update, the names of the APIs in the module is lost, and in its place, the text switches to "XML library" and talks about factories and processors. I understand the focus of this PR is to introduce a system property to allow for configuration file but it's changing important API docs that introduce the APIs. 

My view is that we have to create a new introduction that provides a summary of the APIs in the module and a short summary of some concepts. We can't have API docs talk about "factories" and "processors" without defining them. This intro/setup is needed before diving into the configuration, which we might title "Configuring factories and processors" with a summary of the various ways that they can be configured. I think the text from the "Property Precedence" section will need to move into the section on Configuration as it is hard to list the various ways to configure without also talking about precedence.

Once you have a good introduction and seciton on Configuration then I think the other sections will follow without too much work.

Reply:
New introduction added, along with sections leading up to the Configuration File.